build.sbt

```diff
@@ -43,6 +44,10 @@ lazy val commonSettings = Seq(
   scalacOptions in (Compile, doc) := (scalacOptions in (Compile, doc)).value.filter(_ != "-Xfatal-warnings")
 ) ++ warnUnusedImport
 
+lazy val machinistDependencies = Seq(
+  libraryDependencies += "org.typelevel" %%% "machinist" % "0.4.1"
```

Comment:
The entire picture isn't that clear to me either. It seems that a later version of simulacrum *does* use imp for summoning typeclasses instead of an `@inline def apply` method.